ckcp/hack/util/reset_ckcp.sh

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SCRIPT_DIR="$(
+  cd "$(dirname "$0")" >/dev/null || exit 1
+  pwd
+)"
+
+PLNSVC_DIR="$(dirname "$(dirname "$(dirname "$SCRIPT_DIR")")")"
+CKCP_DIR="$PLNSVC_DIR/ckcp"
+GITOPS_DIR="$PLNSVC_DIR/gitops"
+CONFIG="$PLNSVC_DIR/config/config.yaml"
+
+RESET_HARD="false"
+
+usage() {
+  printf "
+Usage:
+    %s [options]
+
+Scrap ckcp and free resources deployed by openshift_dev_setup.sh script.
+
+Mandatory arguments:
+    --work-dir WORK_DIR
+        Location of the cluster files related to the environment.
+        A single file with extension kubeconfig is expected in the subdirectory: credentials/kubeconfig/kcp
+        Kubeconfig files for compute clusters are expected in the subdirectory: credentials/kubeconfig/compute
+
+Optional arguments:
+    --reset-hard RESET_HARD
+        Setting this option to 'true' will aggressively remove operators deployed.
+        Defalut value: %s
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    %s --work-dir './tmp/tmp.435kjkdsf'
+" "${0##*/}" "$RESET_HARD" "${0##*/}" >&2
+}
+
+parse_args() {
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+    --work-dir)
+      shift
+      WORK_DIR="$1"
+      ;;
+    --reset-hard)
+      shift
+      RESET_HARD="$1"
+      ;;
+    -d | --debug)
+      set -x
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    --)
+      # End of arguments
+      break
+      ;;
+    *)
+      exit_error "Unknown argument: $1"
+      ;;
+    esac
+    shift
+  done
+}
+
+exit_error() {
+    printf "[ERROR] %s" "$@" >&2
+    usage
+    exit 1
+}
+
+
+prechecks() {
+    WORK_DIR=${WORK_DIR:-}
+    if [[ -z "${WORK_DIR}" ]]; then
+      printf "\n[ERROR] Missing parameter --work-dir" >&2
+      exit 1
+    fi
+    KUBECONFIG="$WORK_DIR/credentials/kubeconfig/compute/compute.kubeconfig.base"
+    if [ ! -f "$KUBECONFIG" ]; then
+      printf "\n[ERROR] Couldn't find compute's kubeconfig." >&2
```

Comment:
It's a good idea to print the expected file path to give users a starting point for debugging. Applicable for the kcp kubeconfig as well.

Reply:
sure